### PR TITLE
workerpool: generic result type for worker pool

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #46258

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/backfilling_scheduler.go [hunk 1]
```diff
@@ -308,12 +308,12 @@ func (b *ingestBackfillScheduler) setupWorkers() error {
 	}
 	b.copReqSenderPool = copReqSenderPool
 	readerCnt, writerCnt := b.expectedWorkerSize()
-	skipReg := workerpool.OptionSkipRegister[idxRecResult]{}
```
Comment (ywqzzy) [inline ddl/backfilling_scheduler.go:311]: Why remove？

Comment (tangenta) [reply ddl/backfilling_scheduler.go:311]: Let's make it the default skip because the feature is immature.

**Review by ywqzzy — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]
The worker pool doesn't need to collect the result in theory, which makes it complex. If we need to know the result, we can wrap these workers as a WorkerWIthResult to collect the result, then include the real worker.

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**